php/stats.php

```diff
@@ -39,101 +41,121 @@
     $sth->execute([$user]);
     $row = $sth->fetch();
     if (!$row) {
-        die('Error;' . _("User not found."));
+        die(json_encode(["error" => ("User not found.")]));
     }
 
     $public = $row["public"];
     if ($public == "N") {
-        die('Error;' . _("This user's flights are not public."));
+        die(json_encode(["error" => ("This user's flights are not public.")]));
     }
 
     $uid = $row["uid"];
 }
 $filter = "f.uid = " . $uid . getFilterString($dbh, $_POST);
-$array = [];
 
 // Convert mi to km if units=K
 $units = $_SESSION["units"] ?? null;
 if ($units == "K") {
-    $unit = _("km");
+    $unit = "km";
     $multiplier = "* " . KM_PER_MILE;
+    $unitMultiplier = KM_PER_MILE;
 } else {
-    $unit = _("mi");
+    $unit = "mi";
     $multiplier = "";
+    $unitMultiplier = 1;
 }
 
+header("Content-type: application/json; charset=utf-8");
+$response = [
+    // All distance values in this message will be of this unit ("km" or "mi")
+    // unless the unit is indicated in the property name.
+    "distance_unit" => $unit,
+    "unique" => [],
+    "extreme" => [],
+    "longshort" => [],
+    "total" => [
+        "segments" => 0
+    ],
+    "average" => [],
+    "by_reason" => [],
+    "by_seattype" => [],
+    "by_mode" => [],
+    "by_class" => [],
+];
+
 // unique airports, and countries
 $sql = "SELECT COUNT(DISTINCT a.apid) AS num_airports, COUNT(DISTINCT a.country) AS num_countries
 FROM flights AS f,airports AS a
 WHERE (f.src_apid = a.apid OR f.dst_apid = a.apid) AND $filter";
 $sth = $dbh->query($sql);
 $row = $sth->fetch();
 if ($row) {
-    $array += $row;
+    $response["unique"]["airports"] = (int)$row["num_airports"];
+    $response["unique"]["countries"] = (int)$row["num_countries"];
 }
 
 // unique airlines (excluding unknown), unique planes, total distance (mi), average distance (localized), average duration
-$sql = "SELECT COUNT(DISTINCT case when alid <> -1 then alid end) AS num_airlines, COUNT(DISTINCT plid) AS num_planes,
-IFNULL(SUM(distance), 0) AS distance, IFNULL(ROUND(AVG(distance) $multiplier),0) AS avg_distance,
-IFNULL(TIME_FORMAT(SEC_TO_TIME(SUM(TIME_TO_SEC(duration))/COUNT(duration)), '%H:%i'), '00:00') AS avg_duration
+$sql = <<<SQL
+SELECT COUNT(DISTINCT case when alid <> -1 then alid end) AS num_airlines,
+       COUNT(DISTINCT plid) AS num_planes,
+       COUNT(*) as segments,
+       IFNULL(SUM(distance), 0) AS distance,
+       IFNULL(AVG(distance), 0) AS avg_distance,
+       IFNULL(SUM(TIME_TO_SEC(duration)), 0) AS duration,
+       IFNULL(TIME_FORMAT(SEC_TO_TIME(SUM(TIME_TO_SEC(duration))/COUNT(duration)), '%H:%i'), '00:00') AS avg_duration
 FROM flights AS f
-WHERE $filter";
+WHERE $filter
+SQL;
 $sth = $dbh->query($sql);
 $row = $sth->fetch();
```

Comment:
We should pass `PDO::FETCH_ASSOC` as the first parameter, as we don't need the 0 based indexed columns anymore.